### PR TITLE
Added more rules for eslint, closes #116, fixed issues that occured by adding new rules

### LINKED ISSUE #116: More rules for linter
After doing #13 I would suggest to expand the linter rules by the following aspects:
- force space after comma in argument list (`f(a, b, c)` instead of `f(a,b,c)`)
- force curly brackets in same line as `if`, `else`, `try`, `catch`, etc.
- no leading/trailing space in brackets (`("abc")` instead of `( "abc" )`
- empty line between functions

### PR DESCRIPTION
**Pull request by realmayus**
closes #116

### CODE REVIEW

File: common/transactionlog/types/UpdatePersonalEvent.ts [hunk 1]
```diff
@@ -24,10 +24,7 @@ export default class UpdatePersonalEvent extends LogUserEvent {
             console.log("key: " + key);
 
             // Add all new keys
-            if (oldKeys.indexOf(key) == -1) diff[key] = newObj[key];
-            // Add all keys that changed
-            else if (oldObj[key] != newObj[key]) diff[key] = newObj[key];
-            else console.log("unchanged");
+            if (oldKeys.indexOf(key) == -1) { diff[key] = newObj[key]; } else if (oldObj[key] != newObj[key]) { diff[key] = newObj[key]; } else { console.log("unchanged"); } // Add all keys that changed
```
Comment (Jonasdoubleyou) [inline common/transactionlog/types/UpdatePersonalEvent.ts:27]: The automatic formatting seems to create very long lines. Maybe those can be moved into multiple lines?

File: common/entity/Course.ts [hunk 2]
```diff
@@ -110,21 +110,18 @@ export class Course {
     guests: CourseGuest[];
 
     async updateCourse(update: ApiCourseUpdate) {
-        if (!update.isValid())
-            throw new Error("Cannot use invalid ApiCourseUpdate to update course!");
+        if (!update.isValid()) { throw new Error("Cannot use invalid ApiCourseUpdate to update course!"); }
 
-        if (update.instructors)
-            update.instructors = await Promise.all(update.instructors.map(it => getManager().findOneOrFail(Student, { where: { id: it.id, isInstructor: true }})));
+        if (update.instructors) { update.instructors = await Promise.all(update.instructors.map(it => getManager().findOneOrFail(Student, { where: { id: it.id, isInstructor: true }}))); }
```
Comment (Jonasdoubleyou) [inline common/entity/Course.ts:115]: here too

File: common/util/subjectsutils.ts [hunk 3]
```diff
@@ -27,9 +27,9 @@ export function toPupilSubjectDatabaseFormat(s: Subject) {
 }
 
 function parseJSONObjectFormatGradeArray(parsedArray: any[]) {
-    return parsedArray.flatMap( s => {
-        if (typeof s.name !== "string") return [];
-        if ((typeof s.minGrade === "number" && typeof s.maxGrade !== "number") || (typeof s.maxGrade === "number" && typeof s.minGrade !== "number")) return [];
+    return parsedArray.flatMap(s => {
+        if (typeof s.name !== "string") { return []; }
+        if ((typeof s.minGrade === "number" && typeof s.maxGrade !== "number") || (typeof s.maxGrade === "number" && typeof s.minGrade !== "number")) { return []; }
```
Comment (Jonasdoubleyou) [inline common/util/subjectsutils.ts:32]: here too

File: web/controllers/certificateController/index.ts [hunk 4]
```diff
@@ -69,35 +69,25 @@ export async function createCertificateEndpoint(req: Request, res: Response) {
         const { pupil, automatic, endDate, subjects, hoursPerWeek, hoursTotal, medium, activities, ongoingLessons } = req.body;
         const requestor = res.locals.user as Student;
 
-        if (requestor instanceof Pupil)
-            return res.status(403).send("Only students may request certificates");
+        if (requestor instanceof Pupil) { return res.status(403).send("Only students may request certificates"); }
 
-        if (!pupil || !endDate || !subjects || hoursPerWeek === undefined || hoursTotal === undefined || !medium || !activities)
-            return res.status(400).send("Missing parameters");
+        if (!pupil || !endDate || !subjects || hoursPerWeek === undefined || hoursTotal === undefined || !medium || !activities) { return res.status(400).send("Missing parameters"); }
```
Comment (Jonasdoubleyou) [inline web/controllers/certificateController/index.ts:74]: in this paragraph too

File: web/controllers/certificateController/index.ts [hunk 5]
```diff
@@ -273,31 +254,23 @@ export async function signCertificateEndpoint(req: Request, res: Response) {
 
     const entityManager = getManager();
 
-    if (typeof certificateId !== "string")
-        return res.status(400).send("Missing parameter certificateId");
+    if (typeof certificateId !== "string") { return res.status(400).send("Missing parameter certificateId"); }
 
-    if (signaturePupil !== undefined && (typeof signaturePupil !== "string" || !signaturePupil.match(VALID_BASE64)))
-        return res.status(400).send("Parameter signaturePupil must be a string and valid base64 encoding");
+    if (signaturePupil !== undefined && (typeof signaturePupil !== "string" || !signaturePupil.match(VALID_BASE64))) { return res.status(400).send("Parameter signaturePupil must be a string and valid base64 encoding"); }
```
Comment (Jonasdoubleyou) [inline web/controllers/certificateController/index.ts:259]: here too

**Comment by realmayus** [comment]
@Jonasdoubleyou Thanks for the review! I added a rule that restricts the number of statements per line, as well as one that requires newlines in chained function calls. This resolved many of the excessive line length issues.

**Review by Jonasdoubleyou — DISMISSED** [review]

**Review by Jonasdoubleyou — APPROVED** [review]

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**